lib/SPIRV/libSPIRV/SPIRVValue.h

```diff
@@ -98,8 +98,34 @@ class SPIRVValue : public SPIRVEntry {
 
   void setAlignment(SPIRVWord);
   void setVolatile(bool IsVolatile);
-  void setNoSignedWrap(bool HasNoSignedWrap);
-  void setNoUnsignedWrap(bool HasNoUnsignedWrap);
+
+  template <spv::Decoration NoIntegerWrapDecoration>
+  void setNoIntegerDecorationWrap(bool HasNoIntegerWrap) {
+    if (!HasNoIntegerWrap) {
+      eraseDecorate(NoIntegerWrapDecoration);
+      return;
+    }
+    // NoSignedWrap and NoUnsignedWrap decorations are available only if it is
+    // allowed to use SPIR-V 1.4 or if SPV_KHR_no_integer_wrap_decoration
+    // extension is enabled
+    const std::string InstStr =
```

Comment:
The last moment here is that maybe it would be better to define InstStr only when SPIRVDBG is defined.

Reply:
Applied, thanks!